huaweicloud/services/acceptance/waf/resource_huaweicloud_waf_dedicated_instance_test.go

```diff
@@ -0,0 +1,187 @@
+)
+
+func TestAccWafDedicatedInstanceV1_basic(t *testing.T) {
+	var instance instances.DedicatedInstance
+	resourceName := "huaweicloud_waf_dedicated_instance.instance_1"
+	name := acceptance.RandomAccResourceName()
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.TestAccPreCheck(t) },
+		Providers:    acceptance.TestAccProviders,
+		CheckDestroy: testAccCheckWafDedicatedInstanceV1Destroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccWafDedicatedInstanceV1_conf(name),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckWafDedicatedInstanceV1Exists(resourceName, &instance),
+					resource.TestCheckResourceAttr(resourceName, "name", name),
+					resource.TestCheckResourceAttr(resourceName, "available_zone", "cn-north-4b"),
+					resource.TestCheckResourceAttr(resourceName, "cpu_architecture", "x86"),
+					resource.TestCheckResourceAttr(resourceName, "specification_code", "waf.instance.professional"),
+					resource.TestCheckResourceAttr(resourceName, "security_group.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "run_status", "1"),
+					resource.TestCheckResourceAttr(resourceName, "access_status", "0"),
+					resource.TestCheckResourceAttr(resourceName, "upgradable", "0"),
+					resource.TestCheckResourceAttrSet(resourceName, "server_id"),
+					resource.TestCheckResourceAttrSet(resourceName, "service_ip"),
+					resource.TestCheckResourceAttrSet(resourceName, "subnet_id"),
+					resource.TestCheckResourceAttrSet(resourceName, "vpc_id"),
+					resource.TestCheckResourceAttrSet(resourceName, "cpu_flavor"),
+				),
+			},
+			{
+				Config: testAccWafDedicatedInstanceV1_update(name),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckWafDedicatedInstanceV1Exists(resourceName, &instance),
+					resource.TestCheckResourceAttr(resourceName, "name", name+"_updated"),
+					resource.TestCheckResourceAttr(resourceName, "available_zone", "cn-north-4b"),
+					resource.TestCheckResourceAttr(resourceName, "cpu_architecture", "x86"),
+					resource.TestCheckResourceAttr(resourceName, "specification_code", "waf.instance.professional"),
+					resource.TestCheckResourceAttr(resourceName, "security_group.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "run_status", "1"),
+					resource.TestCheckResourceAttr(resourceName, "access_status", "0"),
+					resource.TestCheckResourceAttr(resourceName, "upgradable", "0"),
+					resource.TestCheckResourceAttrSet(resourceName, "server_id"),
+					resource.TestCheckResourceAttrSet(resourceName, "service_ip"),
+					resource.TestCheckResourceAttrSet(resourceName, "subnet_id"),
+					resource.TestCheckResourceAttrSet(resourceName, "vpc_id"),
+					resource.TestCheckResourceAttrSet(resourceName, "cpu_flavor"),
+				),
+			},
+			{
+				ResourceName:            resourceName,
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{},
```

Comment:
you can delete this line as the array is empty.

Reply:
deleted